### PR TITLE
Added login API and unittests.

### LINKED ISSUE #123: Create API route for logging in
Called by:
- The login interface

Functionality:
- If the correct information got entered, send jwt to user

Restrictions:
- logged out users only

### PR DESCRIPTION
**Pull request by ddhanesha**
Resolves #123 

## Test Path
To test the API first create a superuser to create different users. 
Run this command to create superuser: `./manage.py createsuperuser` 

Run server and go to `/admin` to open django admin console. Enter username and password. 
Now, go to User table and click on `add_user` on the top right side. 
Now, to test login API go to `localhost:8000/login` and add this data to login
```
{
   "username": >username<,
   "password": >password<
}
```

Hit Post it will return `token` and `expiry date`. 
Let me know if you have any questions.

### CODE REVIEW

**Comment by ddhanesha** [comment]
I have written LoginAPI based on django rest knox API view. I have also added unittests too.

File: backend/climateconnect_api/factories.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from django.contrib.auth.models import User
+
+
+class UserFactory(factory.DjangoModelFactory):
+    class Meta:
+        model = User
+        django_get_or_create = ('username', 'password', 'is_active')
+
+    is_active = True
+    username="test_user"
```
Comment (positiveimpact) [inline backend/climateconnect_api/factories.py]: Is this a fallback username?

Comment (ddhanesha) [reply backend/climateconnect_api/factories.py]: No, it was for testing only. Removed it. Thanks!

**Comment by ddhanesha** [comment]
Merging this PR!

**Pull request merged**